src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,154 @@
+}
+
+export interface DropdownProps {
+    id?: string;
+    popperProps?: {
+        id: string;
+        modifiers?: PopperProps["modifiers"];
+        "data-popper-id"?: string;
+    };
+    buttonContent?: string;
+    actions: DropdownAction[];
+    children?: React.ReactNode | React.ReactNode[];
+    paperPlacement?: PopperPlacementType;
+    className?: string;
+    disabled?: boolean;
+}
+
+/**
+ *  MUI dropdown component have a default button with dropdown also could be used with
+ * custom button which takes from children, actions array -> array which will be converted
+ * to dropdown menu items.
+ */
+// eslint-disable-next-line react/function-component-definition
+export default function Dropdown({
+    id,
+    actions,
+    buttonContent,
+    popperProps = {
+        id: "popper",
+    },
+    children = null,
+    disabled = false,
+    paperPlacement = "right",
+    className = "",
+}: DropdownProps) {
+    const containerRef = useRef<HTMLDivElement>(null);
+    const [opened, setOpened] = useState(false);
+    const theme = useTheme();
+    const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
+
+    const onClick = useCallback(() => setOpened(true), []);
+    const onClickAway = useCallback(() => setOpened(false), []);
+
+    const onMenuItemClick = useCallback(
+        (actionId: string) => {
+            const targetAction = actions.find((action) => {
+                return action.id === actionId;
+            });
+
+            if (!targetAction) {
+                return;
+            }
+
+            if (!targetAction.shouldMenuStayOpened) {
+                setOpened(false);
+            }
+
+            targetAction.onClick(targetAction);
+        },
+        [actions],
+    );
+
+    const onListKeyDown = useCallback((event: React.KeyboardEvent<Element>) => {
+        if (event.key === "Tab") {
+            event.preventDefault();
+            setOpened(false);
+        }
+    }, []);
+
+    return (
+        <Box className={className} id={id} sx={{ width: isMobile ? "100%" : undefined }}>
+            <div ref={containerRef} onClick={onClick}>
+                {children || (
+                    <DefaultDropdownButton fullWidth={isMobile} disabled={disabled}>
+                        {buttonContent ||
+                            (actions.find(({ isSelected }) => isSelected) || actions[0])?.content}
+                    </DefaultDropdownButton>
+                )}
+            </div>
+            <Popper
+                // @ts-ignore
+                style={{ zIndex: theme.dropdownPopperZindex }}
+                open={opened}
+                anchorEl={containerRef?.current}
+                transition
+                placement={paperPlacement}
+                popperOptions={{ placement: "bottom-start" }}
+                {...popperProps}
+            >
+                {({ TransitionProps }) => (
+                    <Grow {...TransitionProps} style={{ transformOrigin: "center top" }}>
```

Comment:
Why do we need all this stuff?? There's a Dropdown component in cove.js that we should update to deal with the necessary logic